components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -234,19 +229,17 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
   }
 
   convertToRule(): Rule {
-    const conditions: Condition[] = [];
     // This constant ensures type safety
     const equals_op: ConditionOperator = 'EQUALS';
-    this.ruleForm.controls.conditions.value.forEach(c => {
-      conditions.push(<Condition>{
+    const conditions: Condition[] = this.ruleForm.controls.conditions.value.map(
+      (c: {attribute: string, values: string, operator: string}) => ({
         attribute: c.attribute,
         operator: c.operator,
```

Comment:
⚠️ This still is lacking a bit of discipline: `c.operator` is a `string`, `operator` is a string literal type -- we should be checking this properly -- but I also have no idea how to make a mistake appear here at run time...

Reply:
🤔  I'm going to think about that a bit...